python/device/device.py

```diff
@@ -402,49 +409,60 @@ def test_send_message_thread(self, worker_thread_info):
         operations.  Instead, this thread adds the messsage to a queue, and relies on a
-        send_outgoing_messages_thread instance to actually send the message.
+        send_message_thread instance to actually send the message.
         """
 
         while not self.done.isSet():
             worker_thread_info.watchdog_time = time.time()
 
-            pingback_id = str(uuid.uuid4())
-
-            system_health = self.get_system_health_telemetry()
-            longhaul_metrics = self.get_longhaul_metrics()
-
-            props = {"thief": {"cmd": "pingbackRequest", "pingbackId": pingback_id}}
-            props["thief"].update(longhaul_metrics)
-            props["thief"].update(system_health)
-
-            # push these same metrics to Azure Monitor
-            self.send_metrics_to_azure_monitor(props["thief"])
-
-            msg = Message(json.dumps(props))
-            msg.content_type = "application/json"
-            msg.content_encoding = "utf-8"
-            msg.custom_properties["eventDateTimeUtc"] = datetime.datetime.now(
-                datetime.timezone.utc
-            ).isoformat()
-            msg.custom_properties["serviceAppRunId"] = self.service_app_run_id
-
-            def on_pingback_received(pingback_id):
-                self.metrics.send_message_count_received_by_service_app.increment()
+            if self.is_pairing_complete():
+                pingback_id = str(uuid.uuid4())
+
+                system_health = self.get_system_health_telemetry()
+                longhaul_metrics = self.get_longhaul_metrics()
+
+                props = {
+                    "thief": {
+                        "cmd": "pingbackRequest",
+                        "pingbackId": pingback_id,
+                        "serviceAppRunId": self.service_app_run_id,
+                    }
+                }
+                props["thief"].update(longhaul_metrics)
+                props["thief"].update(system_health)
+
+                # push these same metrics to Azure Monitor
+                self.send_metrics_to_azure_monitor(props["thief"])
+
+                msg = Message(json.dumps(props))
+                msg.content_type = "application/json"
+                msg.content_encoding = "utf-8"
+                msg.custom_properties["eventDateTimeUtc"] = datetime.datetime.now(
+                    datetime.timezone.utc
+                ).isoformat()
+                msg.custom_properties["serviceAppRunId"] = self.service_app_run_id
+
+                def on_pingback_received(pingback_id):
+                    self.metrics.send_message_count_received_by_service_app.increment()
+
+                with self.pingback_list_lock:
+                    self.pingback_wait_list[pingback_id] = MessageWaitingToArrive(
+                        callback=on_pingback_received,
+                        pingback_id=pingback_id,
+                        message=msg,
+                        send_time=time.time(),
```

Comment:
is this time UTC?

Reply:
good reminder.  You asked that before and I promised a change.  time.time() returns unix epoch time as an integer (seconds since 1/1/1970).  I think calling it a timestamp might be better?  I'll change this in a future PR.  (I have cascading changes and my deltas get messed up if I'm not careful)